libtransmission/torrent.cc

```diff
@@ -2453,6 +2453,37 @@ bool renameArgsAreValid(std::string_view oldpath, std::string_view newname)
         !tr_strvContains(newname, TR_PATH_DELIMITER);
 }
 
+bool renameNoCollision(tr_torrent const* tor, std::string_view oldpath, std::string_view newname)
+{
+    std::string newpath{};
+    if (tr_strvContains(oldpath, TR_PATH_DELIMITER))
+    {
+        newpath = fmt::format(FMT_STRING("{:s}/{:s}"sv), tr_sys_path_dirname(oldpath), newname);
+    }
+    else
+    {
+        newpath = newname;
+    }
```

Comment:
_(nit)_ `newpath` can const. Also it can be a `tr_pathbuf` to avoid heap allocations

```suggestion
    auto const newpath = tr_strvContains(oldpath, TR_PATH_DELIMITER) ?
        tr_pathbuf{ tr_sys_path_dirname(oldpath), '/', newname } :
        newname;
```